### PR TITLE
Update the Javadoc for all new test suite related classes

### PR DESCRIPTION
**Pull request by big-guy**

### CODE REVIEW

**Comment by big-guy** [comment]
@bot-gradle help

**Comment by big-guy** [comment]
@bot-gradle test this

**Comment by big-guy** [comment]
The commit message here [`e4d4d58` (#18326)](https://github.com/gradle/gradle/pull/18326/commits/e4d4d58b3533e99b8b767b0b507dc0daf57ffe3d) is the best explanation I could come up with for the different meanings of JUnit5, JUnit Platform and JUnit Jupiter.

**Review by tresat — APPROVED** [review]

File: subprojects/testing-jvm/src/main/java/org/gradle/api/tasks/testing/Test.java [hunk 1]
```diff
@@ -956,75 +960,95 @@ TestFramework useTestFramework(TestFramework testFramework) {
...
+     * <p><br>
+     * Use this option if your tests use JUnit Jupiter/JUnit5.
+     * <p><br>
+     * JUnit Platform supports multiple test engines, which allows other testing frameworks to be built on top of it.
+     * You may need to use this option even if you are not using JUnit directly.
+     * <p>
+     * The supplied action configures an instance of {@link org.gradle.api.tasks.testing.junitplatform.JUnitPlatformOptions JUnit Platform specific options}.
      *
-     * @param testFrameworkConfigure An action used to configure the JUnit platform options.
+     * @param testFrameworkConfigure A closure used to configure JUnit platform options.
```
Comment (DPUkyle) [inline subprojects/testing-jvm/src/main/java/org/gradle/api/tasks/testing/Test.java:1019]: Is it correct/convention to use `closure` as an abstract term here? Could this mistakenly be associated with `groovy.lang.Closure`?

Comment (big-guy) [reply subprojects/testing-jvm/src/main/java/org/gradle/api/tasks/testing/Test.java:1019]: ah, I meant to use action here. Thanks for noticing

**Comment by big-guy** [comment]
@bot-gradle test and merge

**Comment by big-guy** [comment]
@bot-gradle test and merge

**Comment by big-guy** [comment]
@bot-gradle cancel

**Comment by big-guy** [comment]
@bot-gradle test and merge

**Comment by big-guy** [comment]
@bot-gradle test and merge

**Pull request merged**